plenum/server/node.py

```diff
@@ -1856,18 +1995,29 @@ def startViewChange(self, proposedViewNo: int):
         :param proposedViewNo: the new view number after view change.
         """
         self.view_change_in_progress = True
+        self._schedule(action=self._check_view_change_completed,
+                       seconds=self._view_change_timeout)
+        self.master_replica.on_view_change_start()
         self.viewNo = proposedViewNo
         logger.debug("{} resetting monitor stats after view change".
                      format(self))
         self.monitor.reset()
-        self.processStashedMsgsForView(proposedViewNo)
+        self.processStashedMsgsForView(self.viewNo)
         # Now communicate the view change to the elector which will
         # contest primary elections across protocol all instances
-        self.elector.viewChanged(self.viewNo)
+        self.elector.view_change_started(self.viewNo)
         self._primary_replica_no = None
         pop_keys(self.msgsForFutureViews, lambda x: x <= self.viewNo)
         self.initInsChngThrottling()
         self.logNodeInfo()
+        # Keep on doing catchup until >2f nodes LedgerStatus same on have a
+        # prepared certificate the first PRE-PREPARE of the new view
+        logger.info('{} changed to view {}, will start catchup now'.
+                    format(self, self.viewNo))
+        # TODO: Do not need to revert, keep the 3PC messages with
+        # prepared certificate
+        # self.master_replica.revert_unordered_batches()
```

Comment:
We will revert unordered state when catch-up is started in any case (there is preStart callback), so no need for these commented lines.